woocommerce/woocommerce-ios
# Update coupon remotely from the Editing screen

Part of #6488 

### Description
As part of the Milestone for editing a coupon, I connected the networking side with the current Edit screen UI. 
Just some notes:
- the `date_expire_gmt` field also if it’s a writable parameter, doesn’t update the expiry date remotely. I think that it’s a bug, because for me the only parameter working is the `date_expire` (not gmt). So I needed to implement a custom decoder in `Coupon` model, for a full conformance (because the CodingKey `dateExpiresNotGmt` will not be used in the decoding phase).
- I implemented a custom encoder for `Coupon`, because previously we was sending the entire body of a Coupon, also some read-only fields including the coupon ID.

### Testing instructions
1. Open a coupon detail and make sure the feature flag for editing a coupon is enabled.
2. Press the ellipsis menu in the navigation menu.
3. Try to edit some fields of the coupon (you can edit basically everything, but not the products and the categories, still a WIP)
4. Save the coupon.
5. You should be able to see the changes applied to the coupon on the web.

---
